ts/state/ducks/conversations.ts

```diff
@@ -514,7 +563,21 @@ function handleMessageExpiredOrDeleted(
     // search if we find this message id.
     // we might have not loaded yet, so this case might not happen
     const messageInStoreIndex = state?.messages.findIndex(m => m.propsForMessage.id === messageId);
+    const editedQuotes = { ...state.quotes };
     if (messageInStoreIndex >= 0) {
+      // Check if the message is quoted somewhere, and if so, remove it from the quotes
+      const msgProps = state.messages[messageInStoreIndex].propsForMessage;
+      const { timestamp, sender } = msgProps;
+      if (timestamp && sender) {
+        const message2Delete = editedQuotes[`${timestamp}-${sender}`];
+        window.log.debug(
+          `Deleting quote {${timestamp}-${sender}} ${JSON.stringify(message2Delete)}`
+        );
+
+        // tslint:disable-next-line: no-dynamic-delete
+        delete editedQuotes[`${timestamp}-${sender}`];
+      }
```

Comment:
why not just to the `delete state.quotes[`${timestamp}-${sender}`];` ? createSlice takes care of doing the diff and I think the conversation redux slice is made with createSlice

Reply:
Unfortunately this is not possible. It generates an error with Immer which links to this https://github.com/immerjs/immer/blob/main/src/utils/errors.ts